File: February/C/Palindromic Substrings/Palindromic Substrings.c
#include<stdio.h>
#include<string.h>
#include<stdio.h>

int countSubstrings(char * s)
{
    int res = 0;
    int left = 0;
    int right = 0;
    int len = strlen(s);
    for (int i = 0; i < len; i++) {
        left = i;
        right = i;
        while (left >= 0 && right < len && s[left] == s[right]) {
            res++;
            left--;
            right++;
        }
        left = i;
        right = i + 1;
        while (left >= 0 && right < len && s[left] == s[right]) {
            res++;
            left--;
            right++;
        }
    }
    return res;
}

File: February/Python/Palindromic Substrings/Palindromic Substrings.py
class Solution:
    def countSubstrings(self, s: str) -> int:
        res =0

        for i in range(len(s)):
            l = r = i
            while l >= 0 and r < len (s) and s[l] == s[r]:
                res +=1
                l -= 1
                r += 1

            l = i
            r = i + 1
            while l >= 0 and r < len (s) and s[l] == s[r]:
                res +=1
                l -= 1
                r += 1
            
        return res
